api/host.go

```diff
@@ -138,3 +143,91 @@ func GetUpdateHostStatusCallback(rq amboy.Queue, status string, notes string, us
 		return httpStatus, modifyErr
 	}
 }
+
+// TerminateSpawnHost is a shared utility function to terminate a spawn host
+func TerminateSpawnHost(ctx context.Context, env evergreen.Environment, h *host.Host, u *user.DBUser, r *http.Request) (*host.Host, int, error) {
```

Comment:
Instead of taking a request as a parameter, this function should let the caller handle what it needs to do with the request. Same with the other ones

Reply:
I pass the request as a parameter so that the log error function can get the necessary information it needs. In this scenario its difficult for the caller to know if the error should be logged or not, Since it can return an error that doesnt need to be logged. So i left that decision to the function itself.